src/Microsoft.AspNet.Mvc.Core/DefaultActionSelector.cs

```diff
@@ -36,86 +36,55 @@ public class DefaultActionSelector : IActionSelector
+            {
+                matchingActions = new List<ActionDescriptor>(matchingActionConstraints.Count);
+                foreach (var candidate in matchingActionConstraints)
                 {
-                    matchingActions = new List<ActionDescriptor>(matchingActionConstraints.Count);
-                    foreach (var candidate in matchingActionConstraints)
-                    {
-                        matchingActions.Add(candidate.Action);
-                    }
+                    matchingActions.Add(candidate.Action);
                 }
+            }
 
-                var finalMatches = SelectBestActions(matchingActions);
-
-                if (finalMatches == null || finalMatches.Count == 0)
-                {
-                    if (_logger.IsEnabled(LogLevel.Verbose))
-                    {
-                        _logger.WriteValues(new DefaultActionSelectorSelectAsyncValues()
-                        {
-                            ActionsMatchingRouteConstraints = matchingRouteConstraints,
-                            ActionsMatchingActionConstraints = matchingActions,
-                            FinalMatches = finalMatches,
-                        });
-                    }
+            var finalMatches = SelectBestActions(matchingActions);
 
-                    return Task.FromResult<ActionDescriptor>(null);
-                }
-                else if (finalMatches.Count == 1)
-                {
-                    var selectedAction = finalMatches[0];
-
-                    if (_logger.IsEnabled(LogLevel.Verbose))
-                    {
-                        _logger.WriteValues(new DefaultActionSelectorSelectAsyncValues()
-                        {
-                            ActionsMatchingRouteConstraints = matchingRouteConstraints,
-                            ActionsMatchingActionConstraints = matchingActions,
-                            FinalMatches = finalMatches,
-                            SelectedAction = selectedAction
-                        });
-                    }
+            if (finalMatches == null || finalMatches.Count == 0)
+            {
+                return Task.FromResult<ActionDescriptor>(null);
+            }
+            else if (finalMatches.Count == 1)
+            {
+                var selectedAction = finalMatches[0];
 
-                    return Task.FromResult(selectedAction);
-                }
-                else
-                {
-                    if (_logger.IsEnabled(LogLevel.Verbose))
-                    {
-                        _logger.WriteValues(new DefaultActionSelectorSelectAsyncValues()
-                        {
-                            ActionsMatchingRouteConstraints = matchingRouteConstraints,
-                            ActionsMatchingActionConstraints = matchingActions,
-                            FinalMatches = finalMatches,
-                        });
-                    }
+                return Task.FromResult(selectedAction);
```

Comment:
Log on all three branches here.

I'd say verbose for a successful match, and an info for no-match